File: gpfksdlrn/level0/수_조작하기1.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181926

function solution(n, control) {
  const commands = {
    w: 1,
    s: -1,
    d: 10,
    a: -10,
  };

  return Array.from(control).reduce(
    (result, command) => result + commands[command],
    n,
  );
}

File: gnsals0904/level0/수_조작하기1.java
class Solution {
    public int solution(int n, String control) {
        for(int i = 0; i < control.length(); i++){
            n += calc(control.charAt(i));
        }
        return n;
    }

    public int calc(char c) {
        return switch (c) {
            case 'w' -> 1;
            case 's' -> -1;
            case 'd' -> 10;
            case 'a' -> -10;
            default -> 0;
        };
    }
}